### ISSUE TITLE
GUI: Add generate method

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#16000]
The `generate` method has been removed from Bitcoin Core, but it would be nice if the GUI had a `generate` method in the debug console. That method would effectively expand to `generatetoaddress(getnewaddress())` (just like the `def generate` in our test node python class)

**Comment by MarcoFalke** [comment]
The same could be added to `bitcoin-cli` (similar to `-getinfo`)

**Comment by piilanismith** [comment]
got it @eklitzke 

how do I put money into bitcoin?

**Comment by piilanismith** [comment]
don't know how

**Comment by piilanismith** [comment]
how do I join gti?

**Comment by fanquake** [comment]
@piilanismith Please stop posting here, questions about how to purchase btc are clearly off topic.

**Comment by jonasschnelli** [comment]
I also miss the `generate` function. Instead of adding it client-side (which would be fine by me), we could add a `generatetowallet(<nblocks>)` function (without directly coupling the miner and the wallet).

**Comment by Meshugah** [comment]
Could I work on this? Also, in general, is there a procedure to starting working on an issue?

**Comment by jonasschnelli** [comment]
@Meshugah: sure. You're more than welcome to work on this. Maybe start by reading the developer notes: https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md

**Comment by ezegom** [comment]
@MarcoFalke Is this issue still open? I'ld like to work on this if it is.

**Comment by brakmic** [comment]
Hi,

I've tried to implement the idea from @MarcoFalke by expanding the bitcoin-cli. However, as I'm not sure if someone else is still working on it, go ahead and just grab my code and integrate it in yours (if it's of any use to you).

--------

There's now the option "-generate" available. The allowed (optional) parameters are *nblocks* and *maxtries*. 
For example:

* bitcoin-cli -generate

```shell
{
  "result": [
    "1e9a6484111cdebe4d1794ad7994a8504e692cdd2c43d8ccea66ce819fbc1209"
  ],
  "error": null,
  "id": 0
}
```

* bitcoin-cli -generate 2

```shell
{
  "result": [
    "2fcd8f7a0b03c0111fec02f185ad20acce68fee8a649b39e6409e51fb6d5dc87",
    "234f63fbdd4d8292617127d4f0559295098785f54372f857a19651d98d2a59ab"
  ],
  "error": null,
  "id": 0
}
```

* bitcoin-cli -generate 3 10000

```shell
{
  "result": [
    "2c5cf3a0ff25b1708b5a4648e4c4c5c09ff309a0afa743f3d602011b97e00be2",
    "0d61891a44ca9f177a4632e4a30d612f78b459cd7dea29ebaa13d647a94c3708",
    "2d1e1e911747d3d5bc452d8a1f8d906971023b9811831aa5b9dd937ef403ab66"
  ],
  "error": null,
  "id": 0
}
```

First, I have tried to implement it directly in wallet/rpcwallet.cpp, but as there is *no good way* to access the server functionality without coupling them together I've moved to bitcoin-cli directly. However, the code I wrote is...abysmal, from my point of view. I works, sure, but maybe there's a nicer, more *Bitcoin-idiomatic* way to accomplish it?

To achieve the results, I had to code two **RequestHandlers** in bitcoin-cli: [GenerateRequestHandler](https://github.com/brakmic/bitcoin/blob/resurrect-generate-method/src/bitcoin-cli.cpp#L244) and [GetNewAddressRequestHandler](https://github.com/brakmic/bitcoin/blob/resurrect-generate-method/src/bitcoin-cli.cpp#L227). These will then be called in [CommandLineRPC](https://github.com/brakmic/bitcoin/blob/resurrect-generate-method/src/bitcoin-cli.cpp#L520). 

Yes, there is some "magic" in between, because I had to manipulate the args vector. This is because I needed a valid address for "generatetoaddress". I'm pretty sure, there are more intelligent ways to do it, so I hope, someone will take this piece of bad code and give it a few antifragile hits. ;)

Here's the commit: https://github.com/brakmic/bitcoin/commit/faf5f037a42bfc01fcaf80985af24ae92c9b7fb0

Regarding **tests**, well, I don't know what would be the proper way to do it. Maybe expanding the Python code (node stuff etc.) by implementing a "generate" call there? If I could get some hints on that, it'd be great. Thanks in advance!

Regards,
